x-pack/plugins/maps/public/shared/layers/sources/wms_source/wms_client.js

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import { parseString } from 'xml2js';
+
+export class WmsClient {
+  constructor({ serviceUrl, version = '1.1.1' }) {
+    this.serviceUrl = serviceUrl;
+    this.version = version;
+  }
+
+  async _fetch(url) {
+    return window.fetch(url);
+  }
+
+  async _fetchCapabilities() {
+    const resp = await this._fetch(`${this.serviceUrl}?version=${this.version}&request=GetCapabilities&service=WMS`);
+    if (resp.status >= 400) {
+      throw new Error(`Unable to access ${this.state.serviceUrl}`);
+    }
+    const body = await resp.text();
+
+    const parsePromise = new Promise((resolve, reject) => {
+      parseString(body, (error, result) => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve(result);
+        }
+      });
+    });
+    return await parsePromise;
+  }
+
+  reduceLayers(path, layers) {
```

Comment:
this doesn't need to be on the prototype. It doesn't access `this`. I'd just put it as helper outside class definition.